quadriga/src/main/java/edu/asu/diging/quadriga/web/DeleteCollectionController.java

```
@@ -22,13 +25,20 @@ public class DeleteCollectionController {
     private Logger logger = LoggerFactory.getLogger(getClass());
     
     @RequestMapping(value = "/auth/collections/{id}/delete", method = RequestMethod.GET)
-    public String get(@PathVariable String id, RedirectAttributes redirectAttributes) {
+    public String get(@PathVariable String id, RedirectAttributes redirectAttributes, Authentication authentication) {
         try {
-            collectionManager.deleteCollection(id);
-            redirectAttributes.addFlashAttribute("alert_type", "success");
-            redirectAttributes.addFlashAttribute("alert_msg", "Collection has been deleted.");
-            redirectAttributes.addFlashAttribute("show_alert", true);
-        
+            SimpleUser simpleUser = (SimpleUser) authentication.getPrincipal();
+            Collection collection = collectionManager.findCollection(id);
+            if (!collection.getUsername().equals(simpleUser.getUsername())) {
```

this should be handled by the collection manager. It should throw an exception if someone else than the owner tries to delete a collection.